internal/proxyservice/proxy_service.go

```diff
@@ -53,3 +157,16 @@ func NewProxyServices(
 
 	return &secretlessObj
 }
+
+// Move this up one level, pass it down for now
+func GetSecrets(secrets []*v2.Credential) (map[string][]byte, error) {
+	providerFactories := make(map[string]func(v1.ProviderOptions) (v1.Provider, error))
+
+	for providerID, providerFactory := range internal.InternalProviders {
+		providerFactories[providerID] = providerFactory
+	}
+
+	resolver := plugin.NewResolver(providerFactories, nil, nil)
+
+	return resolver.Resolve(secrets)
+}
```

Comment:
Not a blocker but this recreates the whole thing for every plugin. We should probably try to figure out how to optimize this by caching it.

Reply:
Agreed.   The whole `GetSecrets` solution here is just temporary to get e2e.  It should be getting injected anyway.  But really as you said we should just revisit it from first principles.  Will add TODO though.

Reply:
TODO missing